includes/class-wc-checkout.php

```diff
@@ -747,7 +747,7 @@ protected function validate_posted_data( &$data, &$errors ) {
 				$field_label = isset( $field['label'] ) ? $field['label'] : '';
 
 				if ( $validate_fieldset &&
-					( isset( $field['type'] ) && 'country' === $field['type'] ) &&
+					( isset( $field['type'] ) && 'country' === $field['type'] && '' !== $data[ $key ] ) &&
```

Comment:
just noting for myself that the value of `$required` variable when the country field is empty string has to be false otherwise it would trip the check in line 823.